plugins/module_utils/role.py

```diff
@@ -53,17 +59,25 @@ def __eq__(self, other):
         )
 
     @classmethod
-    def get_role_from_uuid(cls, role_uuid, rest_client: RestClient, must_exist=False):
+    def get_role_from_uuid(
+        cls, role_uuid: str, rest_client: RestClient, must_exist: bool = False
+    ) -> Union[Role, None]:
         hypercore_dict = rest_client.get_record(
             "/rest/v1/Role/{0}".format(role_uuid), must_exist=must_exist
         )
-        role = cls.from_hypercore(hypercore_dict)
-        return role
+        if hypercore_dict:
```

Comment:
@domendobnikar This is not needed since None case is already handled in the from_hypercore method

Reply:
Oh I forgot to remove the same check from ` from_hypercore` method.
Do we need to raise an exception if `Role` is `None`?

Reply:
It depends on the case, we have "must_exist" parameter that handles this.

Reply:
All from_hypercore methods have None check, so I would leave it there. Or is there any case where it is better to have it in other methods that call from_hypercore method?

Reply:
`Get` methods should handle this. Like you said - we already have `must_exist` implemented there, might as well have the `None` check, so we have everything in one place.
It makes a small difference between:
`from hypercore(cls, hypercore_data: dict[Any, Any])` vs `from hypercore(cls, hypercore_data: Union[dict[Any, Any], None])`
And the return statement:
`-> Union[Role, None]` vs `-> Role`
We just have to pick one and stick to it. I think right now half the classes use it one way and half the other way.
At least the ones that I made. That's why I put it in here, didn't even check or think about it which is my mistake :)

Reply:
Yes you are right, not all have the None check in the from_hypercore method :) But since the majority have it, I would stick to this then.